src/main/java/uk/gov/digital/ho/hocs/casework/application/ContextAwareExecutorDecorator.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.digital.ho.hocs.casework.application;
+
+import org.slf4j.MDC;
+import org.springframework.core.task.TaskExecutor;
+
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+public class ContextAwareExecutorDecorator implements Executor, TaskExecutor {
+
+    private final Executor executor;
+
+    public ContextAwareExecutorDecorator(Executor executor) {
+        this.executor = executor;
+    }
+
+    @Override
+    public void execute(Runnable command) {
+        Runnable ctxAwareCommand = decorateContextAware(command);
+        executor.execute(ctxAwareCommand);
+    }
+
+    private Runnable decorateContextAware(Runnable command) {
```

Comment:
Not sure what this is doing, it looks like it will clear the main thread MDC then replace it again.

Reply:
I (blindly) followed this guide: http://www.chrisport.ch/java/2016/10/12/java-spring-context-aware.html